game-app/game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/general/FiringGroupSplitterGeneral.java

```diff
@@ -56,10 +56,28 @@ public enum Type {
 
   @Override
   public List<FiringGroup> apply(final BattleState battleState) {
+    final Collection<Unit> enemyUnits =
+        CollectionUtils.getMatches(
+            battleState.filterUnits(ALIVE, side.getOpposite()),
+            PredicateBuilder.of(Matches.unitIsNotInfrastructure())
+                .andIf(side == DEFENSE, Matches.unitIsSuicideOnAttack().negate())
+                .andIf(side == OFFENSE, Matches.unitIsSuicideOnDefense().negate())
+                .build());
+
+    // Filter participants (same as is done in MustFightBattle.removeNonCombatants()), so that we
```

Comment:
>    // Filter participants (same as is done in MustFightBattle.removeNonCombatants()), so that we
    // don't end up generating combat step names for units that will be excluded.
    
Is there a reasonable way to unify how combat step names are generated?
Overall though, fixing the two logic paths so that they match again, and so that they call each other out is good. If we don't unify the two pieces of code, we should then add a comment to `MustFightBattle.removeNonCombatants()` to indicate that this location should be updated as well to match.

Reply:
I added a comment to the other location for now.

It *may* be possible to unify things, but it's tricky. Basically, the splitter code does talk to MustFightBattle and gets the units via the `filterUnits()` API. So that could theoretically be made to return the unit with non-combatants removed. But that API is also called from many different places, so such a change may not be safe. Another option would be to make another function next to `filterUnits()` that would do this and call that instead from here, but it's not really saving much duplication.